src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/HeaderFilter.cs

```diff
@@ -0,0 +1,37 @@
+namespace CoreWCF.Dispatcher
+{
+    using System;
+    using CoreWCF.Channels;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Diagnostics;
+    using System.Runtime.Serialization;
+    using CoreWCF.Diagnostics;
+
+    abstract class HeaderFilter : MessageFilter
+    {
```

Comment:
Add internal and move using's outside of namespace. Also that looks like too many using statements so cleanup (VS has a tool to do that)